### PR TITLE
Mask negative notification count

### LINKED ISSUE #7324: -1 Notifications in browser
It's nothing game breaking or something like that but it is definitely a bug so i just want to make it known.
it happened when i open the browser to download some maps after chatting with a friend.
(This was like 4 days ago, so I dont know if it's already fixed)

https://i.imgur.com/T6JjP1d.png
https://i.imgur.com/aLX6i6Q.png

### PR DESCRIPTION
**Pull request by notbakaneko**
closes #7324

### CODE REVIEW

**Review by nanaya — COMMENTED** [review]
Should the `chat-icon` component be updated as well? (why are they even two separate almost same component)

**Comment by nanaya** [comment]
The original issue is even on chat-icon component whoops :eyes:

**Comment by notbakaneko** [comment]
🤔

File: resources/assets/lib/register-components.coffee [hunk 1]
```diff
@@ -84,7 +84,7 @@ reactTurbolinks.register 'comments', CommentsManager, (el) ->
 reactTurbolinks.registerPersistent 'chat-icon', ChatIcon, true, (el) ->
   (try JSON.parse(el.dataset.chatIcon)) ? {}
 
-reactTurbolinks.registerPersistent 'notification-icon', NotificationIcon, true, (el) ->
+reactTurbolinks.registerPersistent 'notification-icon', MainNotificationIcon, true, (el) ->
```
Comment (nanaya) [inline resources/assets/lib/register-components.coffee:87]: the html element class name should probably match the component name

File: resources/assets/lib/notification-icon.tsx [hunk 2]
```diff
@@ -1,53 +1,39 @@
...
+    const modifiers = {
+      glow: this.props.count > 0,
+      mobile: this.props.type === 'mobile',
+    };
+
     return (
-      <span className={this.mainClass()}>
-        <i className='fas fa-inbox' />
+      <span className={classWithModifiers('notification-icon', modifiers)}>
+        <i className='fas fa-comment-alt' />
```
Comment (nanaya) [inline resources/assets/lib/notification-icon.tsx:24]: icon should differ between main and chat

File: resources/assets/lib/notification-icon.tsx [hunk 3]
```diff
@@ -1,53 +1,39 @@
...
-
-    if (this.props.type === 'mobile') {
-      ret += ' notification-icon--mobile';
-    }
-
-    return ret;
-  }
-
   private unreadCountDisplay() {
     if (core.notificationsWorker.hasData) {
```
Comment (nanaya) [inline resources/assets/lib/notification-icon.tsx:33]: I wonder if this should be moved to the chat/main themselves and make this component non-observer

or the other way where the parent component just passing relevant function and this component observes all the changes...

**Review by nanaya — CHANGES_REQUESTED** [review]

**Review by nanaya — APPROVED** [review]

**Pull request merged**